libraries/service_helpers.rb

```diff
@@ -14,34 +14,33 @@ def splunk_bin_path
       Pathname.new(install_dir).join('bin')
     end
 
-    # Returns the process id of the daemon is running
-    # @return [Integer]
+    # Returns the process id of the daemon or nil if is not running.
+    # @return [Integer, nil]
     def service_pid
       install_state = node.run_state['splunk_ingredient']['installations'][install_dir]
       if install_state
         pid_file = Pathname.new(install_dir).join('var/run/splunk/splunkd.pid')
         if pid_file.exist?
-          pid = pid_file.readlines.first.to_i || false
-          return pid
+          pid = pid_file.readlines.first.to_i
+          return pid > 0 ? pid : nil
         end
       end
-      false
     end
 
     # Reports if the daemon for the current package is running
     # @return [Boolean]
     def service_running
-      service_pid != false
+      !!service_pid # rubocop:disable Style/DoubleNegation
```

Comment:
Now that I have tests I can probably do that; I was trying to generally convert to a boolean representing truthiness, not just testing for nil. If false were returned from service_pid, it would return true here.